chat-core/src/OpenAiEmbedFunc.ts

```diff
@@ -44,34 +58,36 @@ export const makeOpenAiEmbedFunc = ({
   );
   url.searchParams.append("api-version", apiVersion);
   return async ({ text, userIp }) => {
-    try {
-      const { data } = await axios.post<CreateEmbeddingResponse>(
-        url.toString(),
-        {
-          input: text,
-          user: userIp,
-        },
-        {
-          headers: {
-            "api-key": apiKey,
-            "Content-Type": "application/json",
+    return backOff(async () => {
```

Comment:
like we discussed in meeting, can you add a note about maybe adding more complex retry logic where it determines if to retry by err type

Reply:
Implemented